packages/trackerless-network/src/logic/StreamPartSplitAvoidance.ts

```diff
@@ -0,0 +1,88 @@
+import { areEqualPeerDescriptors, DhtAddress, getNodeIdFromPeerDescriptor, PeerDescriptor } from '@streamr/dht'
+import { Logger, wait } from '@streamr/utils'
+import { Layer1Node } from './Layer1Node'
+
+/*
+ * Tries to find new neighbors if we currently have less than MIN_NEIGHBOR_COUNT neigbors. It does so by
+ * rejoining the stream's control layer network.
+ * 
+ * This way we can avoid some network split scenarios. The functionality is most relevant for small stream
+ * networks.
+ */
+
+const logger = new Logger(module)
+
+const exponentialRunOff = async (
+    task: () => Promise<void>,
+    description: string,
+    abortSignal: AbortSignal,
+    baseDelay = 500,
+    maxAttempts = 6
+): Promise<void> => {
+    for (let i = 1; i <= maxAttempts; i++) {
+        if (abortSignal.aborted) {
+            return
+        }
+        const factor = 2 ** i
+        const delay = baseDelay * factor
+        try {
+            await task()
+        } catch (e: any) {
+            logger.info(`${description} failed, retrying in ${delay} ms`)
+        }
+        try { // Abort controller throws unexpected errors in destroy?
+            await wait(delay, abortSignal)
+        } catch (err) {
+            logger.trace(`${err}`)  // TODO Do we need logging?
+        }
+    }
+}
+
+export const MIN_NEIGHBOR_COUNT = 4
+
+export interface StreamPartSplitAvoidanceConfig {
+    layer1Node: Layer1Node
+    discoverEntryPoints: (excludedNodes?: Set<DhtAddress>) => Promise<PeerDescriptor[]>
+    exponentialRunOfBaseDelay?: number
+}
+
+export class StreamPartSplitAvoidance {
```

Comment:
Would it make sense to have a word `Network` in the name? Or maybe the functionality is generic, and this could be `NetworkSplitAvoidance`?

Reply:
Rename also fields/variables accordingly

Reply:
which variables would that be?

Reply:
All which have phrase `splitAvoidance` -> `networkSplitAvoidance`. E.g. `streamPartSplitAvoidance` -> `streamPartNetworkSplitAvoidance`.